File: apps/web/lib/types/finance.ts
export type InvoiceType = 'materijal' | 'gorivo' | 'leasing' | 'alati' | 'oprema'

export type Supplier = 'Energy Centar' | 'Pondeljak' | 'Lipa Promet'

export type LeasingCompany = 'Impuls' | 'Unicredit Leasing'

export const INVOICE_TYPE_LABELS: Record<InvoiceType, string> = {
  materijal: 'Materijal',
  gorivo: 'Gorivo',
  leasing: 'Leasing',
  alati: 'Alati',
  oprema: 'Oprema',
}

export const SUPPLIERS: Supplier[] = ['Energy Centar', 'Pondeljak', 'Lipa Promet']

export const LEASING_COMPANIES: LeasingCompany[] = ['Impuls', 'Unicredit Leasing']

export interface FinanceDocMeta {
  original_filename: string
  mime_type: string
  file_size: number
  uploaded_at: string
}

export interface CompanyInvoice {
  id: string
  invoice_type: InvoiceType
  supplier?: string
  leasing_company?: string
  document: FinanceDocMeta
  created_by: string
  created_by_name: string
  created_at: string
}

export interface R1Receipt {
  id: string
  submitted_by: string
  submitter_name: string
  price: number
  document: FinanceDocMeta
  created_at: string
}

File: services/api/dto/finance.go
package dto

import "time"

// ── Invoice types ─────────────────────────────────────────────────────────────

const (
	InvoiceTypeMaterijal = "materijal"
	InvoiceTypeGorivo    = "gorivo"
	InvoiceTypeLeasing   = "leasing"
	InvoiceTypeAlati     = "alati"
	InvoiceTypeOprema    = "oprema"
)

var AllowedInvoiceTypes = map[string]bool{
	InvoiceTypeMaterijal: true,
	InvoiceTypeGorivo:    true,
	InvoiceTypeLeasing:   true,
	InvoiceTypeAlati:     true,
	InvoiceTypeOprema:    true,
}

// ── Suppliers (for materijal) ─────────────────────────────────────────────────

const (
	SupplierEnergyCentar = "Energy Centar"
	SupplierPondeljak    = "Pondeljak"
	SupplierLipaPromet   = "Lipa Promet"
)

var AllowedSuppliers = map[string]bool{
	SupplierEnergyCentar: true,
	SupplierPondeljak:    true,
	SupplierLipaPromet:   true,
}

// ── Leasing companies (for leasing) ──────────────────────────────────────────

const (
	LeasingImpuls           = "Impuls"
	LeasingUnicreditLeasing = "Unicredit Leasing"
)

var AllowedLeasingCompanies = map[string]bool{
	LeasingImpuls:           true,
	LeasingUnicreditLeasing: true,
}

// ── Response DTOs ─────────────────────────────────────────────────────────────

type FinanceDocMeta struct {
	OriginalFilename string    `json:"original_filename"`
	MimeType         string    `json:"mime_type"`
	FileSize         int64     `json:"file_size"`
	UploadedAt       time.Time `json:"uploaded_at"`
}

type CompanyInvoice struct {
	ID             string         `json:"id"`
	InvoiceType    string         `json:"invoice_type"`
	Supplier       *string        `json:"supplier,omitempty"`
	LeasingCompany *string        `json:"leasing_company,omitempty"`
	Document       FinanceDocMeta `json:"document"`
	CreatedBy      string         `json:"created_by"`
	CreatedByName  string         `json:"created_by_name"`
	CreatedAt      time.Time      `json:"created_at"`
}

type R1Receipt struct {
	ID            string         `json:"id"`
	SubmittedBy   string         `json:"submitted_by"`
	SubmitterName string         `json:"submitter_name"`
	Price         float64        `json:"price"`
	Document      FinanceDocMeta `json:"document"`
	CreatedAt     time.Time      `json:"created_at"`
}
